File: data/transcoder_evaluation_gfg/c/LONGEST_COMMON_INCREASING_SUBSEQUENCE_LCS_LIS.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

int f_gold ( int arr1 [ ], int n, int arr2 [ ], int m ) {
  int table [ m ];
  for ( int j = 0;
  j < m;
  j ++ ) table [ j ] = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    int current = 0;
    for ( int j = 0;
    j < m;
    j ++ ) {
      if ( arr1 [ i ] == arr2 [ j ] ) if ( current + 1 > table [ j ] ) table [ j ] = current + 1;
      if ( arr1 [ i ] > arr2 [ j ] ) if ( table [ j ] > current ) current = table [ j ];
    }
  }
  int result = 0;
  for ( int i = 0;
  i < m;
  i ++ ) if ( table [ i ] > result ) result = table [ i ];
  return result;
}


int f_filled ( int arr1 [ ], int n, int arr2 [ ], int m ) {}

int main(void) {
    int n_success = 0;
    int param0_0[] = {1,7,9,35,43,51,51,66,88};
int param0_1[] = {-52,52,-92,-46,-94,30,-36,18,-98,22,-36,96,-88,-50,50};
int param0_2[] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1};
int param0_3[] = {5,74,29};
int param0_4[] = {-84,-74,-70,-62,-56,-56,-52,-2,6,24,28,44,44,52};
int param0_5[] = {0,0,0,1,0,1,0,0,1,1,1,1,0,0,1,0,1,0,1,1,0,0,1,0,1,1,0,0,1,0};
int param0_6[] = {3,4,4,7,15,15,16,22,32,32,37,39,39,41,43,46,47,47,49,75,79,80,86,88,93};
int param0_7[] = {70,-64,0,52,32,-98,38,-8,34,70,98,58,-48,-60,-28,-22,-72,82,-98,-36};
int param0_8[] = {0,0,1,1,1,1,1,1};
int param0_9[] = {46,87,98};
int *param0[10] = {param0_0,param0_1,param0_2,param0_3,param0_4,param0_5,param0_6,param0_7,param0_8,param0_9};
    int param1[] = {5,7,36,1,8,17,19,16,7,2};
    int param2_0[] = {10,21,38,50,65,67,87,93,99};
int param2_1[] = {-58,40,56,-62,-92,-94,40,18,-2,-76,-78,-14,44,84,4};
int param2_2[] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1};
int param2_3[] = {57,33,48};
int param2_4[] = {-98,-96,-88,-66,-32,-26,-24,-20,-4,20,48,74,90,96};
int param2_5[] = {1,0,1,1,0,0,0,0,1,1,1,0,1,0,0,0,0,1,0,0,0,0,1,1,1,1,1,1,1,0};
int param2_6[] = {9,12,15,20,22,27,28,28,30,31,35,39,47,58,58,60,73,74,76,78,80,86,95,96,98};
int param2_7[] = {-18,88,-40,-52,30,-10,-18,-56,84,-22,-64,80,-14,-64,40,92,48,-8,24,82};
int param2_8[] = {0,1,1,1,1,1,1,1};
int param2_9[] = {67,31,54};
int *param2[10] = {param2_0,param2_1,param2_2,param2_3,param2_4,param2_5,param2_6,param2_7,param2_8,param2_9};
    int param3[] = {8,10,22,1,12,15,14,12,7,2};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i],param1[i],param2[i],param3[i]) == f_gold(param0[i],param1[i],param2[i],param3[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}

File: data/transcoder_evaluation_gfg/rust/LONGEST_COMMON_INCREASING_SUBSEQUENCE_LCS_LIS.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut arr1: *mut i32,
                                mut n: i32,
                                mut arr2: *mut i32,
                                mut m: i32) -> i32 {
    let vla = m as usize;
    let mut table: Vec<i32> = ::std::vec::from_elem(0, vla);
    let mut j: i32 = 0 as i32;
    while j < m {
        *table.as_mut_ptr().offset(j as isize) = 0 as i32;
        j += 1
    }
    let mut i: i32 = 0 as i32;
    while i < n {
        let mut current: i32 = 0 as i32;
        let mut j_0: i32 = 0 as i32;
        while j_0 < m {
            if *arr1.offset(i as isize) == *arr2.offset(j_0 as isize) {
                if current + 1 as i32 >
                       *table.as_mut_ptr().offset(j_0 as isize) {
                    *table.as_mut_ptr().offset(j_0 as isize) =
                        current + 1 as i32
                }
            }
            if *arr1.offset(i as isize) > *arr2.offset(j_0 as isize) {
                if *table.as_mut_ptr().offset(j_0 as isize) > current {
                    current = *table.as_mut_ptr().offset(j_0 as isize)
                }
            }
            j_0 += 1
        }
        i += 1
    }
    let mut result: i32 = 0 as i32;
    let mut i_0: i32 = 0 as i32;
    while i_0 < m {
        if *table.as_mut_ptr().offset(i_0 as isize) > result {
            result = *table.as_mut_ptr().offset(i_0 as isize)
        }
        i_0 += 1
    }
    return result;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0_0: [i32; 9] =
        [1 as i32, 7 as i32, 9 as i32,
         35 as i32, 43 as i32, 51 as i32,
         51 as i32, 66 as i32, 88 as i32];
    let mut param0_1: [i32; 15] =
        [-(52 as i32), 52 as i32, -(92 as i32),
         -(46 as i32), -(94 as i32), 30 as i32,
         -(36 as i32), 18 as i32, -(98 as i32),
         22 as i32, -(36 as i32), 96 as i32,
         -(88 as i32), -(50 as i32), 50 as i32];
    let mut param0_2: [i32; 42] =
        [0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32];
    let mut param0_3: [i32; 3] =
        [5 as i32, 74 as i32, 29 as i32];
    let mut param0_4: [i32; 14] =
        [-(84 as i32), -(74 as i32), -(70 as i32),
         -(62 as i32), -(56 as i32), -(56 as i32),
         -(52 as i32), -(2 as i32), 6 as i32,
         24 as i32, 28 as i32, 44 as i32,
         44 as i32, 52 as i32];
    let mut param0_5: [i32; 30] =
        [0 as i32, 0 as i32, 0 as i32,
         1 as i32, 0 as i32, 1 as i32,
         0 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         0 as i32, 0 as i32, 1 as i32,
         0 as i32, 1 as i32, 0 as i32,
         1 as i32, 1 as i32, 0 as i32,
         0 as i32, 1 as i32, 0 as i32,
         1 as i32, 1 as i32, 0 as i32,
         0 as i32, 1 as i32, 0 as i32];
    let mut param0_6: [i32; 25] =
        [3 as i32, 4 as i32, 4 as i32,
         7 as i32, 15 as i32, 15 as i32,
         16 as i32, 22 as i32, 32 as i32,
         32 as i32, 37 as i32, 39 as i32,
         39 as i32, 41 as i32, 43 as i32,
         46 as i32, 47 as i32, 47 as i32,
         49 as i32, 75 as i32, 79 as i32,
         80 as i32, 86 as i32, 88 as i32,
         93 as i32];
    let mut param0_7: [i32; 20] =
        [70 as i32, -(64 as i32), 0 as i32,
         52 as i32, 32 as i32, -(98 as i32),
         38 as i32, -(8 as i32), 34 as i32,
         70 as i32, 98 as i32, 58 as i32,
         -(48 as i32), -(60 as i32), -(28 as i32),
         -(22 as i32), -(72 as i32), 82 as i32,
         -(98 as i32), -(36 as i32)];
    let mut param0_8: [i32; 8] =
        [0 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32];
    let mut param0_9: [i32; 3] =
        [46 as i32, 87 as i32, 98 as i32];
    let mut param0: [*mut i32; 10] =
        [param0_0.as_mut_ptr(), param0_1.as_mut_ptr(), param0_2.as_mut_ptr(),
         param0_3.as_mut_ptr(), param0_4.as_mut_ptr(), param0_5.as_mut_ptr(),
         param0_6.as_mut_ptr(), param0_7.as_mut_ptr(), param0_8.as_mut_ptr(),
         param0_9.as_mut_ptr()];
    let mut param1: [i32; 10] =
        [5 as i32, 7 as i32, 36 as i32,
         1 as i32, 8 as i32, 17 as i32,
         19 as i32, 16 as i32, 7 as i32,
         2 as i32];
    let mut param2_0: [i32; 9] =
        [10 as i32, 21 as i32, 38 as i32,
         50 as i32, 65 as i32, 67 as i32,
         87 as i32, 93 as i32, 99 as i32];
    let mut param2_1: [i32; 15] =
        [-(58 as i32), 40 as i32, 56 as i32,
         -(62 as i32), -(92 as i32), -(94 as i32),
         40 as i32, 18 as i32, -(2 as i32),
         -(76 as i32), -(78 as i32), -(14 as i32),
         44 as i32, 84 as i32, 4 as i32];
    let mut param2_2: [i32; 42] =
        [0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32];
    let mut param2_3: [i32; 3] =
        [57 as i32, 33 as i32, 48 as i32];
    let mut param2_4: [i32; 14] =
        [-(98 as i32), -(96 as i32), -(88 as i32),
         -(66 as i32), -(32 as i32), -(26 as i32),
         -(24 as i32), -(20 as i32), -(4 as i32),
         20 as i32, 48 as i32, 74 as i32,
         90 as i32, 96 as i32];
    let mut param2_5: [i32; 30] =
        [1 as i32, 0 as i32, 1 as i32,
         1 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 0 as i32,
         1 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 1 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 0 as i32];
    let mut param2_6: [i32; 25] =
        [9 as i32, 12 as i32, 15 as i32,
         20 as i32, 22 as i32, 27 as i32,
         28 as i32, 28 as i32, 30 as i32,
         31 as i32, 35 as i32, 39 as i32,
         47 as i32, 58 as i32, 58 as i32,
         60 as i32, 73 as i32, 74 as i32,
         76 as i32, 78 as i32, 80 as i32,
         86 as i32, 95 as i32, 96 as i32,
         98 as i32];
    let mut param2_7: [i32; 20] =
        [-(18 as i32), 88 as i32, -(40 as i32),
         -(52 as i32), 30 as i32, -(10 as i32),
         -(18 as i32), -(56 as i32), 84 as i32,
         -(22 as i32), -(64 as i32), 80 as i32,
         -(14 as i32), -(64 as i32), 40 as i32,
         92 as i32, 48 as i32, -(8 as i32),
         24 as i32, 82 as i32];
    let mut param2_8: [i32; 8] =
        [0 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32];
    let mut param2_9: [i32; 3] =
        [67 as i32, 31 as i32, 54 as i32];
    let mut param2: [*mut i32; 10] =
        [param2_0.as_mut_ptr(), param2_1.as_mut_ptr(), param2_2.as_mut_ptr(),
         param2_3.as_mut_ptr(), param2_4.as_mut_ptr(), param2_5.as_mut_ptr(),
         param2_6.as_mut_ptr(), param2_7.as_mut_ptr(), param2_8.as_mut_ptr(),
         param2_9.as_mut_ptr()];
    let mut param3: [i32; 10] =
        [8 as i32, 10 as i32, 22 as i32,
         1 as i32, 12 as i32, 15 as i32,
         14 as i32, 12 as i32, 7 as i32,
         2 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize], param1[i as usize],
                    param2[i as usize], param3[i as usize]) ==
               f_gold(param0[i as usize], param1[i as usize],
                      param2[i as usize], param3[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
